### PR TITLE
Runtime implementation of array witnesses

### PR DESCRIPTION
**Pull request by aschwaighofer**
(using these to replace the current array witnesses is next)

rdar://27412867
SR-3376

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by aschwaighofer** [comment]
@swift-ci Please test

File: stdlib/public/runtime/Array.cpp [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Config.h"
+#include "swift/Runtime/HeapObject.h"
+#include "swift/Runtime/Metadata.h"
+
+using namespace swift;
+
+enum class ArrayCopy : unsigned {
```
Comment (jrose-apple) [inline stdlib/public/runtime/Array.cpp]: Nitpick: please put these in an anonymous namespace, so they don't cause ODR problems (unlikely but I think possible).

Comment (aschwaighofer) [reply stdlib/public/runtime/Array.cpp]: Good point.

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Pull request merged**